data_model/src/trigger/set.rs

```diff
@@ -411,52 +411,54 @@ impl Set {
 
         let succeed_clone = Arc::clone(&succeed);
         let errors_clone = Arc::clone(&errors);
-        let apply_f = move |id: Id, action: &dyn ActionTrait| {
-            if let Repeats::Exactly(atomic) = action.repeats() {
-                if atomic.get() == 0 {
-                    return;
+
+        {
```

Comment:
Consider extracting into a `fn`

Reply:
Simplified function to avoid using scope or `drop`.